asset/file_reader/index.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const Promise = require('bluebird');
+const Queue = require('@terascope/queue');
+const { getChunk, getOffsets } = require('@terascope/chunked-file-reader');
+const fse = require('fs-extra');
+
+function newSlicer(context, executionContext, retryData, logger) {
+    const opConfig = executionContext.config.operations[0];
+    const queue = new Queue();
+
+    async function processFile(filePath) {
+        const stat = await fse.stat(filePath);
+        const total = stat.size;
+        getOffsets(opConfig.size, total, opConfig.delimiter).forEach((offset) => {
+            offset.path = filePath;
+            queue.enqueue(offset);
+        });
+    }
+
+    async function getFilePaths(filePath) {
+        const dirContents = await fse.readdir(filePath);
```

Comment:
Waiting for confirmation from dev.  Assuming an unhandled exception in a Slicer results in a failed job, I see these options:

1. Let the error go unhandled.
2. Retry with exponential backoff.
3. Catch & log the error.

If we don't want to hang up this PR to implement option 2, then option 1 gets my vote even if means more operator intervention since the times this would be a transient error probably mean there are bigger fires to put out.

Reply:
I think in the short run, letting an error bubble up here is fine. Also, the schema check for the option that gets passed in as `filePath` uses `use.lstat()` and `fse.readdir()` to check if it is a directory with items in it and not a symlink, so the config validation will throw an error otherwise. If a regular directory with empty subdirectories and/or symlinks is provided, the slicer just won't end up generating slices